server/rest/src/main/java/org/infinispan/rest/resources/CacheResourceV2.java

```diff
@@ -134,9 +136,14 @@ public Invocations getInvocations() {
             // Operations
             .invocation().methods(POST).path("/v2/caches/{cacheName}").withAction("clear").handleWith(this::clearEntireCache)
             .invocation().methods(GET).path("/v2/caches/{cacheName}").withAction("size").handleWith(this::getSize)
+
+            // Rolling Upgrade methods
             .invocation().methods(POST).path("/v2/caches/{cacheName}").withAction("sync-data").handleWith(this::syncData)
-            .invocation().methods(POST).path("/v2/caches/{cacheName}").withAction("disconnect-source").handleWith(this::disconnectSource)
-            .invocation().methods(POST).path("/v2/caches/{cacheName}").withAction("connect-source").handleWith(this::connectSource)
+            .invocation().methods(POST).path("/v2/caches/{cacheName}").deprecated().withAction("disconnect-source").handleWith(this::deleteSourceConnection)
+            .invocation().methods(POST).path("/v2/caches/{cacheName}/rolling-upgrade/source-connection").handleWith(this::addSourceConnection)
+            .invocation().methods(DELETE).path("/v2/caches/{cacheName}/rolling-upgrade/source-connection").handleWith(this::deleteSourceConnection)
+            .invocation().methods(HEAD).path("/v2/caches/{cacheName}/rolling-upgrade/source-connection").handleWith(this::hasSourceConnections)
+            .invocation().methods(GET).path("/v2/caches/{cacheName}/rolling-upgrade/source-connection").handleWith(this::getSourceConnection)
```

Comment:
I much prefer this approach :+1: 

As a rolling upgrade connection is always to a source, can we simplify this to `rolling-upgrade/connection`?

Reply:
Not sure, I think connection gives margin for more interpretations